src/transforms/deployment.ts

```diff
@@ -1,7 +1,84 @@
 import _ from "lodash";
-import { Context } from "probot/lib/context";
 import issueKeyParser from "jira-issue-key-parser";
-import { JiraDeploymentData } from "../interfaces/jira";
+import {JiraDeploymentData} from "../interfaces/jira";
+import {GitHubAPI} from "probot";
+import {WebhookPayloadDeploymentStatus} from "@octokit/webhooks";
+import {LoggerWithTarget} from "probot/lib/wrap-logger";
+import {Octokit} from "@octokit/rest";
+import {booleanFlag, BooleanFlags} from "../config/feature-flags";
+
+// https://docs.github.com/en/rest/reference/repos#list-deployments
+async function getLastSuccessfulDeployCommitSha(
+	owner: string,
+	repoName: string,
+	github: GitHubAPI,
+	deployments: Octokit.ReposListDeploymentsResponseItem[],
+	logger?: LoggerWithTarget
+): Promise<string> {
+
+	try {
+		for (const deployment of deployments) {
+			// Get each deployment status for this environment so we can have their statuses' ids
+			const listDeploymentStatusResponse: Octokit.Response<Octokit.ReposListDeploymentStatusesResponse> = await github.repos.listDeploymentStatuses({
+				owner: owner,
+				repo: repoName,
+				deployment_id: deployment.id,
+				per_page: 100 // Default is 30, max we can request is 100
+			});
+
+			// Find the first successful one
+			const lastSuccessful: Octokit.ReposListDeploymentStatusesResponseItem | undefined = listDeploymentStatusResponse.data.find(deployment => deployment.state === "success");
+			if (lastSuccessful !== undefined) {
+				return deployment.sha;
+			}
+		}
+	} catch (e) {
+		logger?.error(`Failed to get deployment statuses.`);
+	}
+
+
+	// If there's no successful deployment on the list of deployments that GitHub returned us (max. 100) then we'll return the last one from the array, even if it's a failed one.
+	return deployments[deployments.length - 1].sha;
+}
+
+async function getCommitMessagesSinceLastSuccessfulDeployment(
+	owner: string,
+	repoName: string,
+	currentDeploySha: string,
+	currentDeployId: number,
+	currentDeployEnv: string,
+	github: GitHubAPI,
+	logger?: LoggerWithTarget
+): Promise<string | undefined> {
+
+	// Grab the last 10 deployments for this repo
+	const deployments: Octokit.Response<Octokit.ReposListDeploymentsResponse> = await github.repos.listDeployments({
+		owner: owner,
+		repo: repoName,
+		environment: currentDeployEnv,
+		per_page: 10 // Default is 30, max we can request is 100
+	})
+
+	// Filter per current environment and exclude itself
+	const filteredDeployments = deployments.data
+		.filter(deployment => deployment.id !== currentDeployId);
+
+	// If this is the very first successful deployment ever, return nothing because we won't have any commit sha to compare with the current one.
+	if (!filteredDeployments.length) {
+		return undefined;
+	}
+
+	const lastSuccessfullyDeployedCommit = await getLastSuccessfulDeployCommitSha(owner, repoName, github, filteredDeployments, logger);
+
+	const commitsDiff = await github.repos.compareCommits({
```

Comment:
Could we extract this out into a function so it could be shared between files? I need to use the exact same function in the solution for builds. No point having the same block on code in both builds and deployments. We can just share 😄

Reply:
Alternatively, I can put it in a shared location in my PR and just update deployments after you get this bad boy merged.

Reply:
The latter sounds like less trouble aka avoiding conflicts :)

Reply:
The PR for builds has been merged. We should update this now to use `compareCommitsBetweenBaseAndHeadBranches` located in `src/transforms/util/githubApiRequests.ts`

Reply:
Think you missed this. We should keep the code dry and reuse where we can. Same logic for this exists in another location. Just need to import it and pass in these same args.

Reply:
Just refactored it and solved conflicts. Next time let's merge older PRs first to avoid conflicts :)